src/NServiceBus.AcceptanceTesting/ContextAppender.cs

```diff
@@ -0,0 +1,171 @@
+﻿namespace NServiceBus.AcceptanceTesting
+{
+    using System;
+    using System.Diagnostics;
+    using Logging;
+
+    /// <summary>
+    /// This class is written under the assumption that acceptance tests are executed sequentially.
+    /// </summary>
+    class ContextAppender : ILoggerFactory, ILog
+    {
+        static object contextLocker = new object();
+
+        void Append(Exception exception)
+        {
+            lock (contextLocker)
+            {
+                Context.Exceptions += exception + Environment.NewLine;
+            }
+        }
+
+        /// <summary>
+        /// Because ILoggerFactory interface methods are only used in a static context. This is the only way to set the currently executing context.
+        /// </summary>
+        /// <param name="newContext">The new context to be set</param>
+        public static void SetContext(ScenarioContext newContext)
+        {
+            lock (contextLocker)
+            {
+                context = newContext;
```

Comment:
I think you are right. It is a leftover from my thinking process about parallel execution. Shall I fix it?